### PR TITLE
[WIP] Create with signature

### PR DESCRIPTION
**Pull request by MoMannn**

### CODE REVIEW

File: src/contracts/tokens/biosample-permission-token.sol [hunk 1]
```diff
@@ -65,4 +66,73 @@ contract BiosamplePermissionToken is
...
+      isValidSignature(
+        address(_tokenId),
+        _claim,
+        _r,
+        _s,
+        _v
+      ),
+      "Signature is not valid."
+    );
+    NFToken._mint(address(_tokenId), _tokenId);
```
Comment (fulldecent) [inline src/contracts/tokens/biosample-permission-token.sol:91]: validate authorized per tokenid namespacing

Comment (MoMannn) [reply src/contracts/tokens/biosample-permission-token.sol:92]: That is automatically check with the signature check.

**Comment by MoMannn** [comment]
@fulldecent This is ready for review.

**Comment by fulldecent** [comment]
For minting, please have this perform two operations -- first minting to the owner -- and then transferring.

This will improve audibility of events.

Gas is not a concern for this project.

**Comment by MoMannn** [comment]
> For minting, please have this perform two operations -- first minting to the owner -- and then transferring.
> 
> This will improve audibility of events.
> 
> Gas is not a concern for this project.

This is for mint or for create with signature? Or both? And if this is for create with signature who does it get created first to? Since anyone can perform this method if he has the signature? And if its for minting I guess mint and transfer does not happen if you are minting to yourself but only mint?

**Comment by fulldecent** [comment]
Here I am only asking that two events be generated for mint.

Therefore we establish an invariant that

For all `Transfer(0, X, Y)` it will be true that `address(Y) == X`.

**Comment by MoMannn** [comment]
@fulldecent Is this how you want it? https://github.com/Genobank/biosample-permission-token/pull/4/commits/317e1ef2aad6c2e2f74b02f482cb379971013b9e

**Comment by fulldecent** [comment]
Perfect. Good to go!

**Pull request merged**